src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+
+<#
+The Initialize-AzMigrateReplicationInfrastructure deploys and configures the replication infrastructure used for server migration in the Azure Migrate project Resource Group.
+Requires: 
+Az.Resources
+Az.KeyVault
+Az.ServiceBus
+Az.Migrate
+
+Example Invocation:  .\Initialize-AzMigrateReplicationInfrastructure.ps1 -ResourceGroupName TestRG  -ProjectName TestProject -Vmwareagentless -TargetRegion centralus
+#>
+
+param(
+    [Parameter(Mandatory)]
+    [System.String]
+    # Specifies the Resource Group of the Azure Migrate Project in the current subscription.
+    ${ResourceGroupName},
+
+    [Parameter(Mandatory)]
+    [System.String]
+    # Specifies the name of the Azure Migrate project to be used for server migration.
+    ${ProjectName},
+
+    [Parameter(Mandatory)]
+    [System.Management.Automation.SwitchParameter]
+    # Specifies the server migration scenario for which the replication infrastructure needs to be initialized.
+    ${Vmwareagentless},
+
+    [Parameter(Mandatory)]
+    [System.String]
+    # Specifies the target Azure region for server migrations.
+    ${TargetRegion},
+
+    [Parameter()]
+    [System.String]
+    # Azure Subscription ID.
+    ${SubscriptionId}
+)
+
+process {
+        Import-Module -Name Az.Migrate
+        # Get/Set SubscriptionId
+        if(($SubscriptionId -eq $null) -or ($SubscriptionId -eq "")){
+            $context = Get-AzContext
+            $SubscriptionId = $context.Subscription.Id
+            if(($SubscriptionId -eq $null) -or ($SubscriptionId -eq "")){
+                throw "Please login to Azure to select a subscription."
+            }
+        }else{
+            Select-AzSubscription -SubscriptionId $SubscriptionId
+        }
+        $context = Get-AzContext
+        Write-Host "Using Subscription Id: ", $SubscriptionId
+        Write-Host "Selected Target Region: ", $TargetRegion
+        
+        $rg = Get-AzResourceGroup -Name $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+        if(!$rg){
+            Write-Host "Creating Resource Group ", $ResourceGroupName
+            New-AzResourceGroup -Name $ResourceGroupName -Location $TargetRegion
+            Write-Host $ResourceGroupName, " created."
+        }
+        Write-Host "Selected resource group : ", $ResourceGroupName
+
+        # Hash code source code
+        $Source = @"
+using System;
+public class HashFunctions
+{
+public static int hashForArtifact(String artifact)
+{
+    int hash = 0;
+    int al = artifact.Length;
+    int tl = 0;
+    char[] ac = artifact.ToCharArray();
+    while (tl < al)
+    {
+        hash = ((hash << 5) - hash) + ac[tl++] | 0;
+    }
+    return Math.Abs(hash);
+}
+}
+"@
+    
+    
+           
+        
+        # Get all appliances and sites in the project
+        $solution = Get-AzMigrateSolution -MigrateProjectName $ProjectName -ResourceGroupName $ResourceGroupName -Name "Servers-Migration-ServerMigration"
```

Comment:
$ResourceGroupName - plz cross check this. target RG name and Azure Migrate project RG name can be different.